pkg/autoscaler/autoscaler/autoscaler_manager.go

```diff
@@ -151,6 +151,9 @@ func (am *autoScalerManager) updateAutoScaling(oldTc *v1alpha1.TidbCluster,
 		return nil, nil
 	}
 
+	// record autoscaler sync timestamp
+	tac.Annotations[label.AnnStsLastSyncTimestamp] = fmt.Sprintf("%d", time.Now().Unix())
```

Comment:
Suggest moving this code to the [Sync](https://github.com/pingcap/tidb-operator/blob/master/pkg/autoscaler/autoscaler/autoscaler_manager.go#L67) function, the sync may return before it reaches to this function.

Reply:
I think the current logic is ok. If the sync ended before it reaches this code, we shouldn't update the tac annotation[AnnStsLastSyncTimestamp]

Reply:
agree with @Yisaer and add one more comment that this code needs happen after caculate last sync time in auto scaler, if this code happens before, we can’t get a real last sync time (always close to when the auto scale happen).

Reply:
I do not understand here, what's this annotation exactly for? What's the standard to update this annotation?

Reply:
This annotation is for tracking last sync/auto scale happen, so that if operator is down for sometime, we can reset phase after operator is up instead of right after operator is up, if we found out the ReadyToScale statisfy threshold and start auto scaling.

Reply:
@vincent178 I think `AnnStsLastSyncTimestamp` is not suitable in this case as this annotation is used to resync the statefulset in periodicity controller. What about using a new annotation named as `AnnLastTikvAutoScaleingSyncTimestamp`

Reply:
Oh, it's useless to update the annotation in Sync function if it cannot reach this function, actually, even if you set the annotation here it may not be updated to the tac because the logic may return before the tac really updated.